include/openssl/ec.h

```diff
@@ -1342,6 +1342,14 @@ void EC_KEY_METHOD_get_verify(const EC_KEY_METHOD *meth,
                                 EVP_PKEY_OP_DERIVE, \
                                 EVP_PKEY_CTRL_GET_EC_KDF_UKM, 0, (void *)(p))
 
+# define EVP_PKEY_CTX_set1_sm2_uid(ctx, uid) \
+        EVP_PKEY_CTX_ctrl(ctx, EVP_PKEY_SM2, EVP_PKEY_OP_TYPE_SIG, \
+                                EVP_PKEY_CTRL_SM2_SET_UID, 0, (void*)(uid))
+
+# define EVP_PKEY_CTX_get1_sm2_uid(ctx, uid) \
```

Comment:
We need to document these two macros too. You should mention that we expect a NUL terminated string for the uid. Also the memory ownership semantics.

Reply:
Either this needs to be get0 and don't use strdup in the implementation, or get1 and use strdup. At the moment its get1 without strdup.

get0 == Reference to the internal data is returned; caller must not free
get1 == A new reference is returned, caller must free
set0 == Ownership of the reference is passed
set1 == Ownership of the reference is not passed (a copy is taken, or reference count is increased)